### DISCUSSION TITLE
URBDRC channel support

### DISCUSSION THREAD

**Discussion by Benyjuice** [neutrinolabs/xrdp#2673] [Ideas]
Freerdp now supports redirecting USB devices from clients to servers through the URBDRC channel. Is it possible to implement URBDRC support for xrdp server using Linux gadget subsystem interface or USBIP kernel module interface?

**Comment by matt335672** [comment]
I'm going to move this to a discussion so that it gets an exposure to a wider audience.

Unless I'm missing something, neither of these interfaces seems designed for this type of operation. It might be possible to get something working, but it is likely to be pretty complex, and as such is unlikely to be something we'd want to take on the maintenance for.

[This project](https://github.com/xairy/raw-gadget) seems more along the lines of what we'd need, but it's not yet of production quality.

I could well be wrong though. Anyone else?
